enhancements/rhcos/rhcos-inject.md

```diff
@@ -0,0 +1,198 @@
+---
+title: rhcos-inject
+authors:
+  - "@crawford"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-10-01
+last-updated: 2020-10-01
+status: provisional
+---
+
+# rhcos-inject
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Throughout it's existence, OpenShift 4 has put a heavy influence on installation flows that make use of installer-provisioned infrastructure. This has largely been successful for predictable environments, such as AWS, Azure, or GCP, but it has (unsurprisingly) proven to make deployments into less predictable environments more difficult. These less predictable environments introduce a lot of variability into areas like network configuration, disk layout, and the life cycle of the machines and that makes it difficult or impossible for OpenShift to start from a foundation of shared assumption. To bridge this gap, admins need a way to inject a certain amount of customization before OpenShift installation can begin.
+
+## Motivation
+
+Admins need a way to inject customization into their RHCOS nodes before they are provisioned by the cluster. In most cases, this configuration is require in order for the provisioning process to complete, so the normal facilities (e.g. Machine Configuration Operator) are not yet available. Today, this is a very manual process involving an admin interactively providing that configuration at the point of running the RHCOS installer. This might work in some environments, but many others don't have easy interactive access due to technical or policy constraints.
+
+### Goals
+
+- Make it easy for an admin to add network configuration to each of the RHCOS nodes added both before and after OpenShift installation.
+- Allow an admin to add custom services and miscellaneous configuration to their RHCOS nodes.
+
+### Non-Goals
+
+- Inventory management
+- Bring-your-own-RHEL customization
```

Comment:
I think we should call out that currently this doesn't solve the problem for IPI baremetal deployments (since that doesn't currently use the RHCOS installer iso), that will be addressed via https://github.com/openshift/enhancements/pull/467